pkg/kubelet/kubelet_pods.go

```diff
@@ -1431,6 +1432,18 @@ func getPhase(spec *v1.PodSpec, info []v1.ContainerStatus) v1.PodPhase {
 		// one container is running
 		return v1.PodRunning
 	case running == 0 && stopped > 0 && unknown == 0:
+		if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
```

Comment:
Without passing a boolean to this method, you don't know whether the pod is actually stopped.  The reason I made the comment in the other PR is that TerminatePod is called with that special knowledge (the containers cannot still be running) which we do not know here because getPhase can be called at multiple points.

Reply:
Let me combine the two PRs.